File: src/Users/Infrastructure/LoginController.php
<?php

declare(strict_types=1);

namespace App\Users\Infrastructure;

use App\Categories\Domain\CustomException;
use App\Shared\Infrastructure\PublicController;
use App\Users\Application\Login\LoginCommand;
use App\Users\Application\Login\LoginHandler;
use JsonException;
use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\JsonResponse;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Routing\Annotation\Route;
use Symfony\Component\Serializer\SerializerInterface;

class LoginController extends AbstractController implements PublicController
{
    public function __construct(
        private readonly LoginHandler $handler,
        private readonly SerializerInterface $serializer
    ) {
    }

    /**
     * @throws JsonException
     */
    #[Route('/login', methods: ['POST'])]
    public function login(Request $request): Response
    {
        $data = json_decode($request->getContent(), true, 512, JSON_THROW_ON_ERROR);
        $command = new LoginCommand(
            $data['email'],
            $data['password'],
        );

        try {
            $result = $this->handler->execute($command);
            return JsonResponse::fromJsonString($this->serializer->serialize($result, 'json'));
        } catch (CustomException $exception) {
            return new Response($exception->getMessage(), $exception->getHttpCode());
        }
    }
}

File: src/main/kotlin/com/zaleort/second_brain/Users/Infrastructure/Controllers/LoginController.kt
package com.zaleort.second_brain.Users.Infrastructure.Controllers

import com.zaleort.second_brain.Users.Application.UseCases.Login.LoginCommand
import com.zaleort.second_brain.Users.Application.UseCases.Login.LoginDTO
import com.zaleort.second_brain.Users.Application.UseCases.Login.LoginHandler
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class LoginController(val handler: LoginHandler) {

    @PostMapping("/api/v1/login")
    fun login(@RequestBody command: LoginCommand): ResponseEntity<LoginDTO> {
        val loginDTO = handler.execute(command)
        return ResponseEntity(loginDTO, HttpStatus.OK)
    }
}